mmv1/third_party/terraform/services/certificatemanager/data_source_google_certificate_manager_certificates.go

```diff
@@ -0,0 +1,267 @@
+package certificatemanager
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+	"google.golang.org/api/certificatemanager/v1"
+)
+
+func DataSourceGoogleCertificateManagerCertificates() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGoogleCertificateManagerCertificatesRead,
+		Schema: map[string]*schema.Schema{
+			"filter": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "global",
+			},
+			"certificates": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
```

Comment:
I wonder if we can use `Schema: datasourceSchemaFromResourceSchema(ResourceCertificateManagerCertificate().Schema),` to mirror the fields in the google_certificate_manager_certificate resource, avoiding the need to manually add new ones.

Reply:
Good suggestion, I was hesitant to do this because there are fields that aren't relevant.
I generated it know and delete the irrelevant fields for the data source.

While I know it's not necessary, I added a remove helper for fields that aren't relevant for the data source.
I prefer having it explicit on fields that can't be part of the data-source instead of having floating fields that exist in the schema but can never be called.